### PR TITLE
Fix manual token creation with plain permalinks

### LINKED ISSUE #4: Manually issuing tokens doesn't work if using default permalinks
```
POST /wp-json/wp/v2/key-pair/1
Status: 404 Not Found
```

Solution is _probably_ to check which permalink method is being used, and use an endpoint like `/?rest_route=/wp-json/wp/v2/key-pair/1`?

### PR DESCRIPTION
**Pull request by jkmassel**
Uses the built-in `get_rest_url` API inside of `get_rest-uri` – this results in permalink-style-aware URI generation.

Fixes #4 

**To Test:**
Before – set a site to use ~default~ plain permalinks. Attempt to create a manual token. It should fail with a network error.
After – set a site to use ~default~ plain permalinks. Attempt to create a manual token. It should succeed.

### CODE REVIEW

**Comment by jkmassel** [comment]
The test is failing here, because the updated `get_rest_uri` function produces absolute URLs, rather than relative ones:

```
1) Test_WP_REST_Key_Pair::test_get_rest_uri
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'/wp-json/wp/v2/key-pair'
+'http://example.org/wp-json/wp/v2/key-pair'
```

IMHO using absolute URIs is the way to go (otherwise how would subdomain multi-site installs work?), so I'd like to update the test. However, I'd love an explicit 👍 from a reviewer on that approach.

**Comment by valendesigns** [comment]
Hi @jkmassel sorry but I don't understand this PR. I currently have default permalinks set on the site I've been developing on, as a matter of fact, I have the entire time I've developed the plugin. I'm not sure I get what the intent is here because what you're saying is broken is working fine for me. I could only assume that there is something I'm either missing in the setup or you have a conflict from another active plugin. Could you explain the steps to reproduce the issue? I have `/%postname%/` set as my permalink structure and I'm able to create a token at this endpoint `/wp-json/wp/v2/token` using postman and an API key/secret without any issue, what am I missing?

File: tests/wp-includes/rest-api/auth/class-test-wp-rest-key-pair.php [hunk 1]
```diff
@@ -107,7 +111,16 @@ public function test_init() {
 	 * @covers ::get_rest_uri()
 	 */
 	public function test_get_rest_uri() {
-		$this->assertEquals( '/wp-json/wp/v2/key-pair', WP_REST_Key_Pair::get_rest_uri() );
+		$this->assertEquals( get_rest_url( null, '/wp/v2/key-pair' ), WP_REST_Key_Pair::get_rest_uri() );
```
Comment (valendesigns) [inline tests/wp-includes/rest-api/auth/class-test-wp-rest-key-pair.php]: `/wp/v2/key-pair` is not a public API, it is meant for the user profile only.

**Comment by jkmassel** [comment]
>I think what you're after is that the plain permalink is not working when you add a new key-pair in the admin. I'll look at it now.

Yep that's right – sorry about the confusion. Updated the initial PR description to hopefully prevent future confusion.

**Comment by jkmassel** [comment]
Closing in favour of #12

**Pull request closed without merging**